tests/test_integration.py

```diff
@@ -958,6 +960,35 @@ async def test_smoke_markdown(envsonschema):
     ), stderr
 
 
+@pytest.mark.asyncio
+async def test_smoke_valid_markdown(envsonschema):
+    stdout, stderr = await bowtie(
+        "smoke",
+        "--format",
+        "markdown",
+        "-i",
+        envsonschema,
+        exit_code=-1,  # because indeed envsonschema gets answers wrong.
+    )
+    parsed_markdown = MarkdownIt("gfm-like", {"linkify": False}).parse(stdout)
+    tokens = SyntaxTreeNode(parsed_markdown).pretty(indent=2)
```

Comment:
Having to test this by going back to a string (i.e. calling `.pretty`) seems a bit non-ideal. Did you check how easy it is to assert against either the thing `.parse` returns directly or else `SyntaxTreeNode`?

Reply:
Parse actually is returning an array of token objects, with nested tokens part of the parent token itself in the children property. Asserting against this would be complex in itself ig thats why I just used the tokens in text form properly separated by indents according to their nested depth.

Output of parse func:

<img width="957" alt="Screenshot 2024-03-18 at 7 00 50 PM" src="https://github.com/bowtie-json-schema/bowtie/assets/75855708/c0d19340-8900-4bfb-9dbe-a18f02b27d6c">

Reply:
Got it, OK, and judging from https://github.com/executablebooks/markdown-it-py/blob/1064835b30325e857252f5b67b34e622261723ca/tests/test_api/test_main.py#L271 that indeed doesn't look like it has any helpers to help us create one. Oh well, that's something missing from the library from what I can tell, but what you did is fine then.